src/map/unit.cpp

```diff
@@ -1608,7 +1621,7 @@ void unit_stop_walking_soon(struct block_list& bl, t_tick tick)
 	}
 	else {
 		// Set coordinates to exact coordinates
-		unit_pos(bl, tick, bl.x, bl.y, ud->sx, ud->sy);
+		ud->getpos(bl.x, bl.y, ud->sx, ud->sy, tick);
```

Comment:
Why do we still need ud->sx and ud->sy here?
I think with your new functions we could even remove the ox/oy caching in this function.